tests/__init__.py

```diff
@@ -1,21 +1,21 @@
 from __future__ import annotations
 
+from pandas._typing import T
 
-def check(
-    actual: object, klass: type, dtype: type | None = None, attr: str = "left"
-) -> None:
+
+def check(actual: T, klass: type, dtype: type | None = None, attr: str = "left") -> T:
 
     if not isinstance(actual, klass):
         raise RuntimeError(f"Expected type '{klass}' but got '{type(actual)}'")
     if dtype is None:
-        return None
+        return actual  # type: ignore[return-value]
```

Comment:
Sounds like a mypy error to me: I think it is of type object because of the instance call where class is just type without a specification which type. Might fallback to object.

Reply:
Opened https://github.com/python/mypy/issues/13462